### ISSUE TITLE
Pleora SDK GST Launch crashes

### ISSUE THREAD

**Issue by NHarishGit** [joshdoe/gst-plugins-vision#8]
I have verified that Pleora eBus player captures images fine on my PC running Win 10 x64 and eBUS v 6.1.1 x 64 bit but when I launch gst (GStreamer v 1.16.1) like below:

"gst-launch-1.0 pleorasrc ! autovideoconvert ! autovideosink"

as soon as gst-launch launches, it crashes with the following error:

"Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
Setting pipeline to PLAYING ...
N*e*w
EcRlRoOcRk::C :G\sUtsSeyrsst\eUmsCelorc\kD
wnloads\gst-plugins-vision-master\gst-plugins-vision-master\sys\pleora\gstpleorasrc.cpp:1619:gst_pleorasrc_create: assertion failed: (minfo.size >= src->pleora_stride * src->height)_"

it seems like, minfo.size is 0 in the above assertion.

What am I missing ?

**Comment by NHarishGit** [comment]
Attached is the snapshot of crash using VS 2017 as d
<img width="1224" alt="gstLaunchCrash" src="https://user-images.githubusercontent.com/57957650/69167309-bed84880-0ac2-11ea-8b89-3707bce8b2fd.png">
ebugger.

**Comment by NHarishGit** [comment]
I think /*Nitin: */ line is missing from the code:

    GST_LOG_OBJECT (src,
        "Row stride not aligned, copying %d -> %d",
        src->pleora_stride, src->gst_stride);

	**### /* Nitin: */
	src->gst_stride = src->pleora_stride;**

    *buf = gst_buffer_new_and_alloc (src->height * src->gst_stride);

**Comment by NHarishGit** [comment]
After fixing like above, I see a warning and a static image from a live camera (not enough buffer queues)

Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
Setting pipeline to PLAYING ...
New clock: GstSystemClock
WARNING: from element /GstPipeline:pipeline0/GstAutoVideoSink:autovideosink0/GstD3DVideoSink:autovideosink0-actual-sink-d3dvideo: Pipeline construction is invalid, please add queues.
Additional debug info:
../libs/gst/base/gstbasesink.c(1218): gst_base_sink_query_latency (): /GstPipeline:pipeline0/GstAutoVideoSink:autovideosink0/GstD3DVideoSink:autovideosink0-actual-sink-d3dvideo:
Not enough buffering available for  the processing deadline of 0:00:00.015000000, add enough queues to buffer  0:00:00.015000000 additional data. Shortening processing latency to 0:00:00.000000000.


How do I fix that ?

**Comment by joshdoe** [comment]
It seems pleorasrc shouldn't even be working with BayerGR8 right now. `src->gst_stride`  is set in `gst_pleorasrc_set_caps`, and it uses `gst_video_info_from_caps` which shouldn't even understand Bayer, it should actually cause the pipeline to fail with an "Unsupported caps..." message. I'd like to see the logs. Can you run it with `GST_DEBUG=pleora*:6` and include it here? That would give some more helpful info. I'm surprised I didn't test it with Bayer, I have access to cameras with that format and the code can already handle it in other places.

As for the one frame coming through and freezing, try doing as the output suggests and add a `queue` to your pipeline. Perhaps the timestamps are no good, try setting `sync=false` to your sink.

**Comment by NHarishGit** [comment]
Josh

I am on a different computer now.

I do have the following configuration:

1. Windows 10 64 bit
2. GStreamer v 1.16.2
3. MS VS 2012 
4. eBUS SDK v 5.1.10.4642
5. GST Plugin version 1.5.92
[640_480_30Hz_Mono8GStPluginLogV1.txt](https://github.com/joshdoe/gst-plugins-vision/files/4284031/640_480_30Hz_Mono8GStPluginLogV1.txt)


Camera is set to 640x480, 30 Hz, Mono8 and I run the following command:

"gst-launch-1.0 -e pleorasrc ! autovideoconvert ! autovideosink"

and attached is debug log level 6 results.

Now, I do NOT see any camera video rendering whereas eBUS player shows the video just fine.

**Comment by joshdoe** [comment]
Ok the log shows the first frame is set to a width and height of 1, then
after that it changes to 640x480. Does it crash or you just don't see
video? Looks like it was producing frames. Try the pipeline again, but try
resizing the small window that pops up. I think I've seen this before,
basically the first frame needs to be dropped. I'll check it out soon, but
try that resize trick first.

**Comment by NHarishGit** [comment]
Thank you for your reply.

While this log is rolling, what I see is a title bar of D3D (Video rendering window) but as soon as I try to resize or maximize it, the application crashes.

to sum it up, I have tried maximizing and tried resizing it, both cause a crash (See attached picture):

![image](https://user-images.githubusercontent.com/57957650/75841872-cd5e1d80-5d9c-11ea-9139-65f574bd437d.png)

**Comment by joshdoe** [comment]
I realized you're running an older (tagged) release, which has the exact problem you discuss. The crash you're seeing only happens with d3dvideosink, which is what autovideosink must be selecting. I can reproduce your crash with d3dvideosink, but if I substitute glimagesink it works fine, so there must be a bug in d3dvideosink for when the window size changes.

This was fixed in https://github.com/joshdoe/gst-plugins-vision/commit/76e67ee5fc399ceb8ea31e26a2d88c6c1132f4a0. I'll close this for now, if that commit doesn't fix your problem let me know.
